app/main/helpers/pagination.py

```diff
@@ -0,0 +1,33 @@
+from urllib.parse import urlparse, parse_qs
+
+
+def get_nav_args_from_api_response_links(links, prev_next_label, request_args, preserved_kwargs):
+    """
+    Get prev/next page numbers from object's API response, plus any extra request args for the URL
+    :param links: dict of links from the API response, e.g. {'self': ..., 'prev': ... 'next': ...}
+    :param prev_next_label: string, either 'prev' or 'next'
+    :param request: the request_args MultiDict with any GET args
+    :param preserved_kwargs: list of string kwargs that should be kept to build the prev/next URLs
```

Comment:
Hey isn't this an opportunity for some type hints? :grin:

Reply:
Ha! (Let's keep that to versioned libraries for now until we get a better idea of how useful/annoying it is)